app/src/main/java/com/example/nigeriatelemedicineapp/registerpatient/RegisterPatientActivity.kt

```diff
@@ -73,4 +76,75 @@ class RegisterPatientActivity : AppCompatActivity() {
         )
     }
 
+    private fun handleStatus(status: Status?) {
+        when (status) {
+
+            Status.EMPTY_FIRST_NAME -> {
+                binding.firstName.requestFocus()
+                binding.firstName.setError("Please enter your first name")
+            }
+
+            Status.EMPTY_LAST_NAME -> {
+                binding.lastName.requestFocus()
+                binding.lastName.setError("Please enter your last name")
+            }
+
+            Status.EMPTY_GENDER -> Toast.makeText(this, "Please select your gender", Toast.LENGTH_SHORT).show()
+
+            Status.EMPTY_PHONE -> {
+                binding.phoneNumber.requestFocus()
+                binding.phoneNumber.setError("Please Enter your phone number")
+            }
+
+            Status.EMPTY_DOB -> {
+                binding.DOB.requestFocus()
+                binding.DOB.setError("Please enter your DOB")
```

Comment:
Are we doing any validation of the DOB beyond just checking for empty string? I think the server will reject any request that isn't formatted correctly. Can we catch that formatting error client-side?